src/selectors/disabled.js

```diff
@@ -37,33 +37,92 @@ export const getNodeDisabledTag = createSelector(
     })
 );
 
+const isNodeOfActiveModularPipeline = (
+  nodeModularPipelines,
+  nodeID,
+  modularPipelineEnabled
+) =>
+  nodeModularPipelines[nodeID].some(
+    (modularPipeline) => modularPipelineEnabled[modularPipeline]
+  );
+
 /**
- * Calculate whether nodes should be disabled based on their modular pipelines
+ * Calculate whether nodes should be disabled based on their modular pipelines,
+ * except related dataset nodes and
  */
 export const getNodeDisabledModularPipeline = createSelector(
   [
     getNodeIDs,
     getModularPipelineEnabled,
     getModularPipelineCount,
     getNodeModularPipelines,
+    getEdgeIDs,
+    getNodeType,
+    getEdgeSources,
+    getEdgeTargets,
   ],
   (
     nodeIDs,
     modularPipelineEnabled,
     modularPipelineCount,
-    nodeModularPipelines
+    nodeModularPipelines,
+    edgeIDs,
+    nodeType,
+    edgeSources,
+    edgeTargets
   ) =>
     arrayToObject(nodeIDs, (nodeID) => {
       if (modularPipelineCount.enabled === 0) {
         return false;
       }
-      if (nodeModularPipelines[nodeID].length) {
-        // Hide task nodes that don't have at least one modular pipeline filter enabled
-        return !nodeModularPipelines[nodeID].some(
-          (modularPipeline) => modularPipelineEnabled[modularPipeline]
+      if (
+        !isNodeOfActiveModularPipeline(
+          nodeModularPipelines,
+          nodeID,
+          modularPipelineEnabled
+        ) &&
+        (nodeType[nodeID] === 'parameters' || nodeType[nodeID] === 'data')
+      ) {
+        const relatedEdgeIDs = edgeIDs.filter((edgeID) =>
+          edgeID.includes(nodeID)
         );
+
+        let isMPEdge = false;
+
+        relatedEdgeIDs.forEach((relatedEdgeID) => {
+          const source = edgeSources[relatedEdgeID];
+          const target = edgeTargets[relatedEdgeID];
+
+          const isInput =
+            source === nodeID &&
+            isNodeOfActiveModularPipeline(
+              nodeModularPipelines,
+              target,
+              modularPipelineEnabled
+            );
+
+          const isOutput =
+            target === nodeID &&
+            isNodeOfActiveModularPipeline(
+              nodeModularPipelines,
+              source,
+              modularPipelineEnabled
+            );
+
+          // check if the target node belongs to a enabled modualr pipeline
+          if ((isInput || isOutput) && isMPEdge === false) {
+            isMPEdge = true;
+          }
+        });
+        return !isMPEdge;
       }
-      return true;
+
+      // Hide nodes that don't have at least one modular pipeline filter enabled
+      return !isNodeOfActiveModularPipeline(
```

Comment:
This seems duplicated with the earlier `if` check. Should we short circuit by moving it up instead?

```js
if (!isNodeOfActiveModularPipeline(
      nodeModularPipelines,
      nodeID,
      modularPipelineEnabled
    ) || (nodeType[nodeID] === 'task') {
   return false;
}
```

Reply:
Thanks for the suggestions - I had a look into this, and while the arrangement of the returned statement would stand as is (i.e we still need to return the result of the check and this cannot be combined with the check for the task node), I had extracted the check for `isNodeOfActiveModularPipeline` out to a variable so the check will not be duplicated and allows more readability. I had also added further comments to explain the exceptions that the `if` statements filters out before returning the result of this check as the final returned value.